action.yml

```diff
@@ -49,9 +52,32 @@ runs:
         # the request metadata to pkg.julialang.org when installing
         # packages via `Pkg.test`.
         JULIA_PKG_SERVER: ""
+    - name: Install dependencies in their own (shared) environment
+      run: |
+        if VERSION > v"1.8pre"
+          using Pkg
+          Pkg.activate("tests-logger-env"; shared=true)
+          Pkg.add(Pkg.PackageSpec(name="GitHubActions", version="0.1"))
+        end
+      shell: julia --color=yes {0}
+      if: inputs.annotate == 'true'
     - run: |
         # The Julia command that will be executed
-        julia_cmd=( julia --color=yes --depwarn=${{ inputs.depwarn }} --inline=${{ inputs.inline }} --project=${{ inputs.project }} -e 'import Pkg;include(joinpath(ENV["GITHUB_ACTION_PATH"], "kwargs.jl"));kwargs = Kwargs.kwargs(;coverage = :(${{ inputs.coverage }}),force_latest_compatible_version = :(${{ inputs.force_latest_compatible_version }}), julia_args = ["--check-bounds=${{ inputs.check_bounds }}"]);Pkg.test(; kwargs...)' )
+        julia_cmd=( julia --color=yes --depwarn=${{ inputs.depwarn }} --inline=${{ inputs.inline }} --project=${{ inputs.project }} -e '
+        import Pkg
+        include(joinpath(ENV["GITHUB_ACTION_PATH"], "kwargs.jl"))
+        kwargs = Kwargs.kwargs(;coverage = :(${{ inputs.coverage }}),force_latest_compatible_version = :(${{ inputs.force_latest_compatible_version }}), julia_args = ["--check-bounds=${{ inputs.check_bounds }}"])
+
+        if ${{ inputs.annotate }} && VERSION > v"1.8pre"
+          push!(LOAD_PATH, "@tests-logger-env") # access dependencies
+          using GitHubActions, Logging
+          global_logger(GitHubActionsLogger())
+          include(joinpath(ENV["GITHUB_ACTION_PATH"], "test_wrapper.jl"))
+          pop!(LOAD_PATH)
+          TestWrapper.test(; kwargs...)        
+        else
+          Pkg.test(; kwargs...)        
+        end' )
```

Comment:
At this length, this should probably be a separate file instead of using `-e`

Reply:
That's not so easy because we are interpolating inputs from the action here

Reply:
We could set them as environment variables and then use `ENV["CHECK_BOUNDS"]` with

```yaml
env:
  CHECK_BOUNDS: ${{ inputs.check_bounds }}
```

Reply:
Ok, done